homeassistant/components/ozw/climate.py

```diff
@@ -278,6 +289,14 @@ async def async_set_hvac_mode(self, hvac_mode):
             _LOGGER.warning("Received an invalid hvac mode: %s", hvac_mode)
             return
         hvac_mode_value = HVAC_MODE_ZW_MAPPINGS.get(hvac_mode)
+        if hvac_mode == HVAC_MODE_HEAT_COOL:
+            # determine the correct value to set
+            for val in self.values.mode.value[VALUE_LIST]:
+                zwave_mode = val[VALUE_ID]
+                hass_mode = ZW_HVAC_MODE_MAPPINGS.get(zwave_mode)
+                if hass_mode == HVAC_MODE_HEAT_COOL:
+                    hvac_mode_value = zwave_mode
+                    break
```

Comment:
Can't we build a reverse lookup table from the beginning and save it as an instance attribute?

Reply:
> Can't we build a reverse lookup table from the beginning and save it as an instance attribute?

sure, fixed in last commit

Reply:
> Can't we build a reverse lookup table from the beginning and save it as an instance attribute?

Sure, fixed in last commit